nixos/modules/services/hardware/asusctl.nix

```diff
@@ -0,0 +1,482 @@
+          standard = ["Static", "Breathe", "Pulse"]
+          multizone = []
+          per_key = false
+
+          [[led_data]]
+          prod_family = "ROG Zephyrus G14"
+          board_names = ["GA402R"]
+          standard = ["Static", "Breathe", "Pulse", "Rainbow"]
+          multizone = []
+          per_key = false
+
+          # GA503QE at higher priority (first match) than GA503Q
+          [[led_data]]
+          prod_family = "ROG Zephyrus G15"
+          board_names = ["GA503QE"]
+          standard = ["Static", "Breathe", "Pulse"]
+          multizone = []
+          per_key = false
+
+          [[led_data]]
+          prod_family = "ROG Zephyrus G15"
+          board_names = ["GA503Q", "GA503R"]
+          standard = ["Static", "Breathe", "Pulse", "Rainbow", "Strobe"]
+          multizone = []
+          per_key = false
+
+          [[led_data]]
+          prod_family = "ROG Zephyrus"
+          board_names = ["GX550L"]
+          standard = ["Static", "Breathe", "Strobe", "Rainbow", "Star", "Rain", "Highlight", "Laser", "Ripple", "Pulse", "Comet", "Flash"]
+          multizone = []
+          per_key = true
+
+          [[led_data]]
+          prod_family = "ROG Zephyrus Duo 15 SE"
+          board_names = ["GX551Q"]
+          standard = ["Static", "Breathe", "Strobe", "Rainbow", "Pulse"]
+          multizone = []
+          per_key = true
+
+          [[led_data]]
+          prod_family = "ROG Flow X13"
+          board_names = ["GV301QH", "GV301QE"]
+          standard = ["Static", "Breathe", "Pulse"]
+          multizone = []
+          per_key = false
+
+          [[led_data]]
+          prod_family = "ROG Strix"
+          board_names = ["G513IC", "G513RC"]
+          standard = ["Static", "Breathe", "Strobe", "Rainbow", "Pulse"]
+          multizone = []
+          per_key = false
+
+          [[led_data]]
+          prod_family = "ROG Flow X16"
+          board_names = ["GV601RW", "GV601RM"]
+          standard = ["Static", "Breathe", "Strobe", "Pulse"]
+          multizone = []
+          per_key = false
+        '';
+      };
+
+      package = mkPackageOption pkgs "asusctl" {
+        default = [ "asusctl" ];
+      };
+
+      autoStartRogControlCenter = mkOption {
+        default = true;
+        type = types.bool;
+        description = lib.mdDoc "Whether rog-control-center should be started automatically.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [
+      cfg.package
+      (mkIf cfg.autoStartRogControlCenter (pkgs.makeAutostartItem { name = "rog-control-center"; package = cfg.package; }))
+    ];
+
+    systemd.services.asusd = {
+      description = "ASUS Notebook Control";
+      wantedBy = [ "multi-user.target" ];
+      before = [ "multi-user.target" ];
+      startLimitIntervalSec = 200;
+      startLimitBurst = 2;
+
+      serviceConfig = {
+        Environment = "IS_SERVICE=1";
+        ExecStartPre = "${pkgs.coreutils}/bin/sleep 2";
```

Comment:
Can we use some condition instead?

Reply:
copied it from the repo but agreed that is is not good so I removed it.